Gems/Atom/RHI/Code/Tests/TagRegistryTests.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RHITestFixture.h"
+#include <Atom/RHI/TagRegistry.h>
+#include <Atom/RHI/TagBitRegistry.h>
+#include <Atom/RHI.Reflect/Handle.h>
+
+namespace UnitTest
+{
+    using namespace AZ;
+
+    class TagRegistryTest
+        : public RHITestFixture
+    {
+    public:
+        TagRegistryTest()
+            : RHITestFixture()
+        {}
+
+        static constexpr uint8_t Count = 4;
+        using IndexType = uint16_t;
+        using TagType = RHI::Handle<IndexType>;
```

Comment:
Could this be expressed as `using TagType = RHI::TagRegistry<IndexType, 32>::TagType`

Reply:
It could, and I thought about that, but decided it was just a more verbose way of saying basically the same thing. In the unlikely event that the TagType changes in TagRegistry, this could just be updated. I don't feel strongly at all about this though so if you think it'd be much better to do `RHI::TagRegistry<IndexType, 32>::TagType` I can change it.